programs/programs/nc/nc_lieap/calculator.py

```diff
@@ -20,19 +24,69 @@ class NCLieap(ProgramCalculator):
     def household_eligible(self, e: Eligibility):
         household_size = self.screen.household_size
 
+        #  set earned deduction based on senior/disabled status
+        earned_deduction = 0 if self._has_senior_or_disabled() else NCLieap.earned_deduction
+
         # has rent, mortgage or heating expense
         has_program_expense = self.screen.has_expense(NCLieap.expenses)
         e.condition(has_program_expense)
 
-        # income
-        gross_income = self.screen.calc_gross_income("yearly", ["all"])
-        income_limit = int(self.fpl_percent * self.program.year.as_dict()[household_size])
-        e.condition(gross_income < income_limit, messages.income(gross_income, income_limit))
+        # Get child care and child support(paid) expenses (yearly)
+        if household_size > 1:
```

Comment:
Can you explain this condition? I wonder if it is necessary to add this household size condition in conjunction with the childcare_expenses list? I would guess we could simplify to just:

`childcare_expenses = self.screen.calc_expenses("yearly", ["childCare", "childSupport"])`

What do you think?

Reply:
Thanks @SonaliBedge, I agree that logically a household without a child wouldn't have childcare expenses. But I don't think it is the calculator's responsibility to enforce that. The calculator's job should be to calculate benefits based on provided data, not to validate business rules about what expenses are "allowed." That type of input validation belongs in the frontend/API layer which is outside the scope of this ticket.